docs/adr/0008_signing_verification.md

```diff
@@ -0,0 +1,358 @@
+# ADR-0008: Digest Calculation & Signing/Pinning (OCI Cosign Support)
+
+- **Status:** Proposed
+- **Deciders:** OCM Maintainers
+- **Date:** 2025-08-27
+- **Related:** OCM signing specification and examples
+
+---
+
+## Context
+
+Component Descriptors (CDs) are signed without altering digest fields during signing. Digests for resources, component references, and the component version are produced in a dedicated step and embedded into the descriptor. Signing uses fixed canonicalization (JCS, RFC 8785) over the descriptor to compute the component-version digest and create a signature envelope. When a component version references another, the digest of the referenced descriptor is embedded in the parent’s `references[]` and contributes to the parent’s component-version digest.
+
+Some workflows require an additional **OCI Cosign** signature over the same canonical payload, optionally uploaded to an OCI registry and recorded in a transparency log (Rekor).
+
+---
+
+## Decision
+
+### Two-step flow
+
+1. **Digest calculation** mutates the descriptor to embed required digests (resources, component references, component-version).
+2. **Signing** appends a signature envelope without computing or embedding digests. Optionally, the same payload may be **cosigned**.
+
+### Commands (verb–noun, long options only, no stdin)
+
+```bash
+# Step 1 — Digest calculation (mutates descriptor: digests only)
+ocm add digests cv <ref> --recurse --force
+
+# Step 2 — Signing (no digest mutation)
+# Direct signing via a single --cert (+ optional password).
+# Optional OCI Cosign signing for the same canonical payload.
+ocm sign cv <ref> \
```

Comment:
this flag set is completely against our hard coding of signing plugins. we explicitly do not wanted to introduce hard coded flags for signing implementations on the commandline which is why https://github.com/open-component-model/ocm-project/issues/544 exists. Does this here suggest dropping this again?

Reply:
applied plugin stuff as far as I understood it